app/models/master_file.rb

```diff
@@ -175,7 +175,8 @@ def destroy
 
     # Stops all processing
     if workflow_id.present? && !finished_processing?
-      encoder_class.find(workflow_id).cancel!
+      encoding_job = encoder_class.find(workflow_id)
+      encoding_job.cancel! if encoding_job
```

Comment:
you can make this cleaner by using ruby's `.try` method if you want just FYI

```ruby
encoder_class.find(workflow_id).try(:cancel!)
```
or if you want to use ruby 2.3 syntax the `&.` also does this

```ruby
encoder_class.find(workflow_id)&.(:cancel!)
```
more info here: http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

Route issue though is `.find()` behaviour unlike ActiveRecord. Assuming this is activefedora or something? ActiveRecord would have raised an RecordNotFound exception on a bad find instead of returning nil